Automattic/wp-calypso
# Gifting: Add blog option for enable / disable

#### Proposed Changes

Add toggle for Gifting subscriptions.

#### Tasks
- [x] Must be behind a flag until https://github.com/Automattic/payments-shilling/issues/1165 is completed. (flag: `subscription-gifting`)
- [ ] ~The toggle only will be visible for users with AUTO RENEW OFF (as soon as they turn it off) (If the user turns AUTO RENEW ON, we will hide the toggle)~ tracked here: https://github.com/Automattic/dotcom-forge/issues/1136
- [x] The toggle will be ON by default users can turn it OFF.
- [x] Save setting & track event on toggle click


> **Warning**
> ~Still WIP, the toggle actually doesn't save the option.~
> ~For now worked only with simple sites~ See updated testing instructions

#### Testing Instructions

Simple sites

1. Make sure you are sandboxed.
2. Apply this diff: D90818-code
3. Go to `/settings/general/[your-site]?flags=subscription-gifting` using a site with a paid plan.
4. If Auto-renew is enabled in `/purchases/subscriptions`, you should not see this toggle after `Footer credit`.
5. Turn off Auto-renew in `/purchases/subscriptions` and go back to settings. You should see the toggle.

Atomic sites
1. Checkout to [this branch](https://github.com/Automattic/jetpack/pull/27137) if not merged yet
2. Sync jetpack to a WoA test site: pb5gDS-1rQ-p2
3. Go to `/settings/general/[your-site]?flags=subscription-gifting` using a site with a paid plan.
4. Try toggling and saving the settings, they should persist

#### Screenshot

https://user-images.githubusercontent.com/6586048/198674572-14435c33-7ef7-4e23-a6a2-853fbd2c06f1.mp4

<img width="776" alt="image" src="https://user-images.githubusercontent.com/402286/198083193-3e7cfb46-fb5c-4cfa-b449-59100ff19199.png">

#### Pre-merge Checklist




Fixes https://github.com/Automattic/dotcom-forge/issues/1081
